pf/tfbridge/packed_provider_info.go

```diff
@@ -0,0 +1,70 @@
+package tfbridge
+
+import (
+	"github.com/json-iterator/go"
+	pfmuxer "github.com/pulumi/pulumi-terraform-bridge/pf/internal/muxer"
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge"
+	shim "github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim"
+	shimUtil "github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim/util"
+)
+
+func UnpackProviderInfo(jsonProviderInfo []byte, bareProvider shim.Provider) (*tfbridge.ProviderInfo, error) {
+	var marshalled WithProviderAliases
+	jsoni := jsoniter.ConfigCompatibleWithStandardLibrary
+	err := jsoni.Unmarshal(jsonProviderInfo, &marshalled)
+	if err != nil {
+		return nil, err
+	}
+	info := marshalled.Info.Unmarshal()
+	info.P = bareProvider
+
+	// Set the maps inside the provider
+	for a, t := range marshalled.ResourceAliases {
+		info.P.ResourcesMap().AddAlias(a, t)
+	}
+	for a, t := range marshalled.DatasourceAliases {
+		info.P.DataSourcesMap().AddAlias(a, t)
+	}
+
+	return info, nil
+}
+
+func PackProviderInfo(info *tfbridge.ProviderInfo) ([]byte, error) {
+	var resources shimUtil.AliasingResourceMap
+	var datasources shimUtil.AliasingResourceMap
+	if p, ok := info.P.(*shimUtil.AliasingProvider); ok {
+		resources = p.ResourceMap
+		datasources = p.DataSourceMap
+	} else if p, ok := info.P.(*pfmuxer.ProviderShim); ok {
+		resources = p.ResourcesMap().(shimUtil.AliasingResourceMap)
+		datasources = p.DataSourcesMap().(shimUtil.AliasingResourceMap)
+	} else {
+		panic("can't find aliases from this provider")
+	}
```

Comment:
This is ugly, but lets us discover aliases without adding a `RangeAliases` method to the base `shim.Provider`